### ISSUE TITLE
Organising classical music

### ISSUE THREAD

**Issue by sondr3** [sampsyo/beets#1364]
Hey, I'm having some slight troubles keeping my classical music organised. I've been organising most my music with `$albumartist` which works great for everything but classical because you end up with folders named like `Пётр Ильич Чайковский; Koninklijk Concertgebouworkest, Antal Doráti/` when I would only want the composer to be included (`Пётр Ильич Чайковский/`). I tried doing this `%if{$composer,$composer,$albumartist}/` to see if it worked but it stills looks like it gives them `$albumartist` names instead.

**Comment by sampsyo** [comment]
We don't currently fetch the composer, sadly: see #506. :cry: Contributions to help get this field would be greatly appreciated.

**Comment by sondr3** [comment]
I really have no idea how you'd to that honestly, but would it be possible to use the Inline plugin to remove everything after the first `;` occurs? That'd be the duct tape solution for me at least.

**Comment by sampsyo** [comment]
Sure. Something like `myartist: albumartist.split(';')[0]`.

**Comment by sondr3** [comment]
That seems to do the job, thanks :grinning:

**Comment by sondr3** [comment]
Sorry, I'm back again, I've been trying to create another item field for the rest of the albumartists but I just can't get it to work. It either works for albums where there's only one albumartist or for albums where there are multiple, I just can't get it to work with either. So far I have:

```
restartist:
  if len(albumartist) != 1:
    pass
  elif len(albumartist) >= 1:
    return albumartist.split(';')[1]
```

I've tried a bunch of variations, checking if it's greater than/smaller than/equal/etc and reversing whether it checks first and so on. But either I get an `IndexError: list index out of range` or it just doesn't do anything. Any ideas (pretty new to Python)?

**Comment by sampsyo** [comment]
`len(albumartist) >= 1` just means you have at least one *character* in the artist string. Maybe you want:

    parts = albumartist.split(';', 1)
    if len(parts) > 1:
        return parts[1]
    else:
        return ''

**Comment by sondr3** [comment]
It works! I thought albumartist was a list, which is probably where I messed up. But regardless, thank you so much for all the help! :D

**Comment by NErnstI** [comment]
Hi!

Thanks for the neat write-down for a non python-savy user like me. - An issue I encountered with this setup is when Discogs is used instead of Musicbrainz. Albumartists are then often called: "composer - conductor, orchestra..."

How can I split the albumartist both by ";" and " - " (including the whitespaces!)

**Comment by jackwilsdon** [comment]
@NErnstI Which field are you trying to extract? The composer?

**Comment by NErnstI** [comment]
Sorry, let me rephrase:

I assumed that the above behaviour is still the case, at least I found no composer tags after importing with beets. So I did this exact workaround:

`item_fields:
    mainartist: albumartist.split(';')[0]
    restartists: |
      parts = albumartist.split(';', 1)
      if len(parts) > 1:
        return parts[1]
      else:
        return ''
`
I then use mainartist as the albumartist mainfolder name (in this case the composer usually) and include the remaining artists in the album subfolder.

When importing from musicbrainz it works with the semicolon, however discogs appears to format it differently, instead of "composer; conductor, orchestra, ..." its "composer - conductor, orchestra, ...".

I would like the above ruleset to apply to discogs format as well, splitting the albumartist both with ";" and " - ".

**Comment by jackwilsdon** [comment]
Try this;

```yaml
item_fields:
    mainartist: |
      import re
      re.split(';|,', albumartist)[0]
    restartists: |
      import re
      parts = re.split(';|,', albumartist)
      if len(parts) > 1:
        return parts[1]
      else:
        return ''
```

**Comment by NErnstI** [comment]
Ah, I see, I already stumpled upon this re-option, but wasn't confident enough to try it myself.

Worked like a charm, just adapted it a bit and separated at the " - " string

`item_fields:
    mainartist: |
      import re
      x = re.split(';| - ', albumartist)
      return x[0]
    restartists: |
      import re
      parts = re.split(';| - ', albumartist)
      if len(parts) > 1:
        return parts[1]
      else:
        return ''
`
Thank you very much for you help :)

**Comment by dosoe** [comment]
Hi! For organising classical music, you might be interested in the parentwork plugin: https://github.com/beetbox/beets/pull/2580#pullrequestreview-44197490
